lib/openssl/signature_algorithm/eddsa.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require "ed25519"
+require "openssl/signature_algorithm/base"
+
+module OpenSSL
+  module SignatureAlgorithm
+    class EdDSA < Base
+      class Ed25519
+        class SigningKey < ::Ed25519::SigningKey
+          def verify_key
+            VerifyKey.new(keypair[32, 32])
+          end
+        end
+
+        class VerifyKey < ::Ed25519::VerifyKey
+          def self.deserialize(key_bytes)
+            new(key_bytes)
+          end
+
+          def serialize
+            to_bytes
+          end
+        end
+      end
+
+      ACCEPTED_CURVES = ["edwards25519"].freeze
+
+      attr_reader :curve
+
+      def initialize(curve: self.class::ACCEPTED_CURVES.first)
+        if self.class::ACCEPTED_CURVES.include?(curve)
+          @curve = curve
+        else
+          raise(OpenSSL::SignatureAlgorithm::UnsupportedParameterError, "Unsupported curve '#{curve}'")
+        end
+      end
```

Comment:
I think we can remove the `curve` keyword argument for now to reduce unnecessary complexity.

We don't know if we are going to support more curves. Maybe, but we don't know yet.

Reply:
Yeah, though about that, but decided to keep the structure to add other curves.

I see your point though, so no problem with removing it.